implementations/exec.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+
+from code_df import (
+    code_changes_git,
+    code_changes_lines,
+    reviews,
+    reviews_accepted,
+    )
+
+from scripts import (
+    code_changes_git,
+    code_changes_lines,
+    reviews,
+    reviews_accepted)
+
+import utils
+
+
+Git_metrics = [code_changes_git, code_changes_lines]
+GitHub_metrics = [reviews, ]
+
+
+def write_results(file=None, Git_results=None, GitHub_results=None):
+    print(Git_results)
+    print(GitHub_results)
+
+
+def test_git_metrics(items_file, date_range, is_code=None, conds=None):
+
+    items = utils.read_json_file(items_file)
+    Git_results = {}
+
+    for metric in Git_metrics:
+        metric_obj = metric(items, date_range)
+        Git_results[str(metric)] = metric_obj.compute()
+
+    return Git_results
+
+
+def test_github_metrics(items_file, date_range):
```

Comment:
This method looks really similar to `test_git_metrics`. Probably, it could be generalized. For instance, instead of having a list of metrics for each data source (see lines 19-20), we could have a dict that relates data sources to metrics like this one:
```
{
  'git': [code_changes_git, code_changes_lines],
  'github': [reviews]
}
```

The map will be read in the `main`, for each key (e.g., `git`, `github`), the corresponding Perceval backend will be initialized and executed. The obtained items, together with the values of the dict (e.g., `[code_changes_git, code_changes_lines]`) will be passed to the generalized version of `test_git_metrics`, and the results returned. WDYT?

Reply:
Please see my comment after this review.